x-pack/plugins/maps/public/shared/layers/sources/es_search_source/es_search_source.js

```diff
@@ -254,9 +336,25 @@ export class ESSearchSource extends AbstractESSource {
     const featureCollection = sourceDataRequest ? sourceDataRequest.getData() : null;
     const meta = sourceDataRequest ? sourceDataRequest.getMeta() : {};
 
+    if (this._isTopHits()) {
+      const entitiesFoundMsg = i18n.translate('xpack.maps.esSearch.topHitsEntitiesCountMsg', {
+        defaultMessage: `Found {entityCount} entities.`,
+        values: { entityCount: meta.entityCount }
+      });
+      if (meta.areResultsTrimmed) {
+        const trimmedMsg = i18n.translate('xpack.maps.esSearch.topHitsResultsTrimmedMsg', {
+          defaultMessage: `Results limited to most recent {topHitsSize} documents per entity.`,
+          values: { topHitsSize: this._descriptor.topHitsSize }
+        });
+        return `${entitiesFoundMsg} ${trimmedMsg}`;
+      }
+
+      return entitiesFoundMsg;
+    }
+
     if (meta.areResultsTrimmed) {
       return i18n.translate('xpack.maps.esSearch.resultsTrimmedMsg', {
-        defaultMessage: `Results limited to first {count} matching documents.`,
+        defaultMessage: `Results limited to first {count} documents.`,
         values: { count: featureCollection.features.length }
```

Comment:
Looks like `featureCollection` could default to null if there's no `sourceDataRequest`. Maybe use a lodash get?